jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushContainerConfigurationStep.java

```diff
@@ -16,90 +16,58 @@
 import com.google.cloud.tools.jib.builder.ProgressEventDispatcher;
 import com.google.cloud.tools.jib.builder.TimerEventDispatcher;
 import com.google.cloud.tools.jib.configuration.BuildConfiguration;
 import com.google.cloud.tools.jib.hash.Digests;
+import com.google.cloud.tools.jib.http.Authorization;
 import com.google.cloud.tools.jib.image.Image;
 import com.google.cloud.tools.jib.image.json.ImageToJsonTranslator;
 import com.google.cloud.tools.jib.json.JsonTemplate;
-import com.google.common.util.concurrent.ListenableFuture;
-import com.google.common.util.concurrent.ListeningExecutorService;
 import java.io.IOException;
 import java.util.concurrent.Callable;
-import java.util.concurrent.ExecutionException;
 
 /** Pushes the container configuration. */
-class PushContainerConfigurationStep
-    implements AsyncStep<AsyncStep<PushBlobStep>>, Callable<AsyncStep<PushBlobStep>> {
+class PushContainerConfigurationStep implements Callable<BlobDescriptor> {
 
   private static final String DESCRIPTION = "Pushing container configuration";
 
   private final BuildConfiguration buildConfiguration;
-  private final ListeningExecutorService listeningExecutorService;
   private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
 
-  private final AuthenticatePushStep authenticatePushStep;
-  private final BuildImageStep buildImageStep;
-
-  private final ListenableFuture<AsyncStep<PushBlobStep>> listenableFuture;
+  private final Authorization pushAuthorization;
+  private final Image builtImage;
 
   PushContainerConfigurationStep(
-      ListeningExecutorService listeningExecutorService,
       BuildConfiguration buildConfiguration,
       ProgressEventDispatcher.Factory progressEventDispatcherFactory,
-      AuthenticatePushStep authenticatePushStep,
-      BuildImageStep buildImageStep) {
-    this.listeningExecutorService = listeningExecutorService;
+      Authorization authenticatePushStep,
+      Image builtImage) {
     this.buildConfiguration = buildConfiguration;
     this.progressEventDispatcherFactory = progressEventDispatcherFactory;
-    this.authenticatePushStep = authenticatePushStep;
-    this.buildImageStep = buildImageStep;
-
-    listenableFuture =
-        AsyncDependencies.using(listeningExecutorService)
-            .addStep(buildImageStep)
-            .whenAllSucceed(this);
+    this.pushAuthorization = authenticatePushStep;
+    this.builtImage = builtImage;
   }
 
   @Override
-  public ListenableFuture<AsyncStep<PushBlobStep>> getFuture() {
-    return listenableFuture;
-  }
-
-  @Override
-  public AsyncStep<PushBlobStep> call() throws ExecutionException {
-    ListenableFuture<PushBlobStep> pushBlobStepFuture =
-        AsyncDependencies.using(listeningExecutorService)
-            .addStep(authenticatePushStep)
-            .addStep(NonBlockingSteps.get(buildImageStep))
-            .whenAllSucceed(this::afterBuildConfigurationFutureFuture);
-    return () -> pushBlobStepFuture;
```

Comment:
Confirming: this core logic can run only after all of the followings are ready:
1. push Authorization
1. the target `Image` instance built